third_party/forked/kubernetes/pkg/proxy/userspace/loadbalancer_ex.go

```diff
@@ -412,37 +416,44 @@ func (lb *LoadBalancerEX) OnDestinationRuleAdd(dr *istioapi.DestinationRule) {
 	lb.lock.Lock()
 	defer lb.lock.Unlock()
 
-	svcName := types.NamespacedName{Namespace: dr.Namespace, Name: dr.Name}
-	if _, exists := lb.strategyMap[svcName]; !exists {
-		lb.setLoadBalancerStrategy(dr, getStrategyName(dr))
+	strategyName := getStrategyName(dr)
+	for svcPort, state := range lb.services {
+		if svcPort.Namespace != dr.Namespace || svcPort.Name != dr.Name {
```

Comment:
It is recommended to use a function to do this code, e.g. isNamespacedNameEqual(xxx)
```
		if svcPort.Namespace != dr.Namespace || svcPort.Name != dr.Name {
			continue
		}
```